discovery_engine_core/core/Cargo.toml

```diff
@@ -26,4 +26,6 @@ xayn-discovery-engine-providers = { path = "../providers" }
 claim = "0.5.0"
 float-cmp = "0.9.0"
 mockall = "0.11.0"
+serde = { version = "1.0.136", features = ["derive"] }
```

Comment:
`serde` is not needed as an additional dev-dependency because it is already a dependency with the same features.
```suggestion
```